### ISSUE TITLE
[bug] Cannot build v2.2.2 with Python 3.12

### ISSUE THREAD

**Issue by gjasny** [conan-io/conan#16076]
### Describe the bug

Hello,

I tried to build conan on the Ubuntu 24.04 beta which includes Python 3.12. When running `python pyinstaller.py` the *first* time on the venv it fails with:

```
10.68 6378 INFO: Looking for Python shared library...
10.71 Traceback (most recent call last):
10.71   File "/testcase/conan/pyinstaller.py", line 158, in <module>
10.71     output_folder = pyinstall(src_folder, args.onefile)
10.71                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
10.71   File "/testcase/conan/pyinstaller.py", line 144, in pyinstall
10.71     PyInstaller.__main__.run(command_args)
10.71   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/__main__.py", line 212, in run
10.71     run_build(pyi_config, spec_file, **vars(args))
10.71   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/__main__.py", line 69, in run_build
10.71     PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
10.71   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/building/build_main.py", line 1186, in main
10.72     build(specfile, distpath, workpath, clean_build)
10.72   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/building/build_main.py", line 1126, in build
10.72     exec(code, spec_namespace)
10.72   File "/testcase/conan/conan.spec", line 8, in <module>
10.72     a = Analysis(
10.72         ^^^^^^^^^
10.72   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/building/build_main.py", line 527, in __init__
10.72     self.__postinit__()
10.72   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/building/datastruct.py", line 184, in __postinit__
10.72     self.assemble()
10.72   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/building/build_main.py", line 637, in assemble
10.72     python_lib = bindepend.get_python_library_path()
10.72                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
10.72   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/depend/bindepend.py", line 815, in get_python_library_path
10.72     python_libname = findLibrary(name)
10.72                      ^^^^^^^^^^^^^^^^^
10.72   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/depend/bindepend.py", line 570, in resolve_library_path
10.72     fullpath = _resolve_library_path_unix(name)
10.72                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
10.72   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/depend/bindepend.py", line 641, in _resolve_library_path_unix
10.72     utils.load_ldconfig_cache()
10.72   File "/testcase/venv/lib/python3.12/site-packages/PyInstaller/depend/utils.py", line 326, in load_ldconfig_cache
10.72     from distutils.spawn import find_executable
10.72 ModuleNotFoundError: No module named 'distutils'
```

But if I run `python pyinstaller.py` another time, all subsequent calls succeed.

Could you please take a look?

Thanks,
Gregor

### How to reproduce it

```
FROM ubuntu:24.04

RUN apt-get update
RUN apt-get upgrade -y
RUN apt-get install -y binutils git python3 python3-dev python3-venv

WORKDIR /testcase
RUN git clone https://github.com/conan-io/conan.git

WORKDIR /testcase/conan

RUN /usr/bin/python3 -m venv /testcase/venv

RUN . /testcase/venv/bin/activate && pip install -e .
#RUN . /testcase/venv/bin/activate && python pyinstaller.py || true
RUN . /testcase/venv/bin/activate && python pyinstaller.py
```

**Comment by memsharded** [comment]
Thanks for reporting @gjasny 

This seems something inside ``PyInstaller`` code, maybe it is not well prepared to run under Python 3.12 yet?

We are aware that Python 3.12 deprecated ``distutils``, we already fixed something in Conan because of this, but maybe PyInstaller suffers something similar

Which version of PyInstaller are you using? Maybe can you try to remove it and install fresh or update it if it is not latest?

**Comment by memsharded** [comment]
Hi @gjasny 

Any feedback here regarding your Pyinstaller version?

**Comment by gjasny** [comment]
Hello,

it uses latest PyInstaller 6.6.0

```
4.043 Successfully installed altgraph-0.17.4 packaging-24.0 pyinstaller-6.6.0 pyinstaller-hooks-contrib-2024.5 setuptools-69.5.1
4.416 143 INFO: PyInstaller: 6.6.0, contrib hooks: 2024.5
4.416 143 INFO: Python: 3.12.3
```

Thanks,
Gregor

**Comment by memsharded** [comment]
I have just tried this combination in Windows:

- Python 3.12
- Pyinstaller 6.6
- Latest Conan from ``develop2`` branch

``python pyinstaller.py`` worked well, this might be limited to Linux only

**Comment by memsharded** [comment]
Submitting https://github.com/conan-io/conan/pull/16186, to remove the installation of ``pyinstaller`` within the script, it will be a precondition.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/16186 merged for next 2.3
